filament/src/View.cpp

```diff
@@ -58,34 +61,37 @@ FView::FView(FEngine& engine)
       mPerViewSb(engine.getPerViewSib()),
       mClipSpace01(engine.getBackend() == Backend::VULKAN),
       mDirectionalShadowMap(engine) {
-    DriverApi& driverApi = engine.getDriverApi();
-
-    mPerViewUbh = driverApi.createUniformBuffer(mPerViewUb.getSize(), driver::BufferUsage::DYNAMIC);
-    mPerViewSbh = driverApi.createSamplerBuffer(mPerViewSb.getSize());
-
-    mPerViewSb.setBuffer(FEngine::PerViewSib::RECORDS, mFroxelizer.getRecordBuffer());
-    mPerViewSb.setBuffer(FEngine::PerViewSib::FROXELS, mFroxelizer.getFroxelBuffer());
+    DriverApi& driver = engine.getDriverApi();
 
+    // set-up samplers
+    mPerViewSb.setBuffer(PerViewSib::RECORDS, mFroxelizer.getRecordBuffer());
+    mPerViewSb.setBuffer(PerViewSib::FROXELS, mFroxelizer.getFroxelBuffer());
     if (engine.getDFG()->isValid()) {
         TextureSampler sampler(TextureSampler::MagFilter::LINEAR);
-        mPerViewSb.setSampler(FEngine::PerViewSib::IBL_DFG_LUT,
+        mPerViewSb.setSampler(PerViewSib::IBL_DFG_LUT,
                 engine.getDFG()->getTexture(), sampler.getSamplerParams());
     }
+    mPerViewSbh = driver.createSamplerBuffer(mPerViewSb.getSize());
+    driver.updateSamplerBuffer(mPerViewSbh, SamplerBuffer(mPerViewSb));
 
-    mIsDynamicResolutionSupported = driverApi.isFrameTimeSupported();
+    // allocate ubos
+    mPerViewUbh = driver.createUniformBuffer(mPerViewUb.getSize(), driver::BufferUsage::DYNAMIC);
+    mLightUbh = driver.createUniformBuffer(CONFIG_MAX_LIGHT_COUNT * sizeof(LightsUib), driver::BufferUsage::DYNAMIC);
 
-    driverApi.updateSamplerBuffer(mPerViewSbh, SamplerBuffer(mPerViewSb));
+    mIsDynamicResolutionSupported = driver.isFrameTimeSupported();
 }
 
 FView::~FView() noexcept = default;
 
 void FView::terminate(FEngine& engine) {
     // Here we would cleanly free resources we've allocated or we own (currently none).
-    DriverApi& driverApi = engine.getDriverApi();
-    driverApi.destroyUniformBuffer(mPerViewUbh);
-    driverApi.destroySamplerBuffer(mPerViewSbh);
-    mDirectionalShadowMap.terminate(driverApi);
-    mFroxelizer.terminate(driverApi);
+    DriverApi& driver = engine.getDriverApi();
+    driver.destroyUniformBuffer(mPerViewUbh);
+    driver.destroyUniformBuffer(mLightUbh);
+    driver.destroySamplerBuffer(mPerViewSbh);
+    driver.destroyUniformBuffer(mRenderableUbh);
```

Comment:
One renderable can live in multiple views, so it is counterintuitive that the per-renderable UIB is "owned" by the view.  Maybe enhance the comment on line 87 to explain this?

Reply:
Ah interesting question. I will update the comment in a later CL, thanks.